### DISCUSSION TITLE
Extracting model parameters after fitting the EVA model

### DISCUSSION THREAD

**Discussion by mubali101** [georgebv/pyextremes#74] [Q&A] [answered]
Hi, thanks for maintaining this incredibly useful package.

After fitting the EVA model with `model.fit_model` it prints a string of output. How can I extract the shape, scale, etc parameters from the output and store it in a variable for later use. Currently, I'm using a crude way to search the string output but surely there must be a neater way to do this...

```python


import  re
test = model
test2 = test.model.__repr__()
# Define regular expressions to extract the values
c_pattern = re.compile(r'c=([-0-9.]+)')
scale_pattern = re.compile(r'scale=([-0-9.]+)')
floc_pattern = re.compile(r'floc=([-0-9.]+)')

# Use regular expressions to find the values in the string
c_match = c_pattern.search(test2)
scale_match = scale_pattern.search(test2)
floc_match = floc_pattern.search(test2)

# Extract the values
c_value = float(c_match.group(1))
scale_value = float(scale_match.group(1))
floc_value = float(floc_match.group(1))
```

Thank you.

**Comment by freyjamo** [comment, marked as answer]
Hi, for the BM parameters, you could check:
nameofyourmodel.model.distribution.mle_parameters['loc']     
nameofyourmodel.model.distribution.mle_parameters['scale']

for the POT:
nameofyourmodel.model.distribution.fixed_parameters['floc']
nameofyourmodel.model.distribution.mle_parameters['scale']
nameofyourmodel.model.distribution.mle_parameters['c']

replace the "nameofyourmodel" with the name that you have given.

Or you can search somewhere similar if you're using another model and not MLE.
Hope this helps, somehow.

**Reply by mubali101** [reply]
Many thanks for your answer. It solved my issue so I'm closing the discussion.
